Providers/Modules/Plugins/ProcessInvestigator/plugin/process_investigator_lib.rb

```diff
@@ -0,0 +1,59 @@
+require_relative 'oms_common'
+require 'securerandom'
+require 'socket'
+
+module OMS
+    class ProcessInvestigator
+
+        def initialize(log)
+            @log = log
+        end
+
+        # ------------------------------------------------------
+        def transform_and_wrap(results)
+            # The limit for event hub messages is 256k, assuming utf16, that's 128k characters.
+            # Leaving some room for overhead, truncate at 100k characters.
+            maxResultLength = 100000
+            record = {}
+            record["PIEventType"] = "Telemetry"
+            record["MachineName"] = (Common.get_hostname or "Unknown Host")
+            record["OSName"] = "Linux " + (Common.get_os_full_name or "Unknown Distro")
+            record["PIVersion"] = "1.19.0930.0003"
+            record["PICorrelationId"] = SecureRandom.uuid
+
+
+            if results.is_a?(Hash)
+                results = results["message"].to_s
+            else
+                results = results.to_s
+            end
+            if results.nil? or results.length == 0
+                @log.error "Process Investigator Filter failed. Empty message."
+                record["PiResults"] = "Process Investigator Filter failed. Empty message."
```

Comment:
sug: follow casing of above elements ["PI..."]

Reply:
Consistency would be nice. Unfortunately there are some down-level dependencies that expect this to be Pi instead of PI.